File: sdk-examples/java/AwsCloudWatchExample.java
// Example: Creating a CloudWatch Alarm using AWS SDK for Java v2

import software.amazon.awssdk.services.cloudwatch.CloudWatchClient;
import software.amazon.awssdk.services.cloudwatch.model.*;

public class AwsCloudWatchExample {
    public static void main(String[] args) {
        CloudWatchClient cloudWatch = CloudWatchClient.create();

        PutMetricAlarmRequest alarmRequest = PutMetricAlarmRequest.builder()
                .alarmName("HighCPUUtilization") // Alarm name
                .metricName("CPUUtilization")
                .namespace("AWS/EC2")
                .statistic(Statistic.AVERAGE)
                .period(300)
                .evaluationPeriods(1)
                .threshold(80.0)
                .comparisonOperator(ComparisonOperator.GREATER_THAN_THRESHOLD)
                .alarmActions("arn:aws:sns:us-west-2:123456789012:my-sns-topic") // Replace with your SNS topic ARN
                .dimensions(Dimension.builder()
                    .name("InstanceId")
                    .value("i-1234567890abcdef0") // Replace with your instance ID
                    .build())
                .build();

        cloudWatch.putMetricAlarm(alarmRequest);

        System.out.println("CloudWatch alarm created.");

        cloudWatch.close();
    }
}

File: sdk-examples/dotnet/AwsCloudWatchExample.cs
// Example: Creating a CloudWatch Alarm using AWS SDK for .NET

using Amazon.CloudWatch;
using Amazon.CloudWatch.Model;
using System;
using System.Threading.Tasks;

class AwsCloudWatchExample
{
    static async Task Main()
    {
        var cloudWatchClient = new AmazonCloudWatchClient();

        var alarmRequest = new PutMetricAlarmRequest
        {
            AlarmName = "HighCPUUtilization", // Alarm name
            MetricName = "CPUUtilization",
            Namespace = "AWS/EC2",
            Statistic = Statistic.Average,
            Period = 300,
            EvaluationPeriods = 1,
            Threshold = 80.0,
            ComparisonOperator = ComparisonOperator.GreaterThanThreshold,
            AlarmActions = new List<string> 
            { 
                "arn:aws:sns:us-west-2:123456789012:my-sns-topic" // Replace with your SNS topic ARN
            },
            Dimensions = new List<Dimension>
            {
                new Dimension
                {
                    Name = "InstanceId",
                    Value = "i-1234567890abcdef0" // Replace with your instance ID
                }
            }
        };

        var response = await cloudWatchClient.PutMetricAlarmAsync(alarmRequest);
        Console.WriteLine("CloudWatch alarm created.");
    }
}